config/suppressions.xml

```diff
@@ -182,4 +182,10 @@
   <!-- the filter is not part of the documentation -->
   <suppress id="settersHaveSinceTag"
             files="[\\/]src[\\/]main[\\/]java[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]XpathFileGeneratorAstFilter.java"/>
+
+  <!-- All xdoc checks/filters files are generated from templates and have escaped characters that
+       make the lines long. We still validate index.xml and template files.
+       until https://github.com/checkstyle/checkstyle/issues/13426 -->
+  <suppress id="lineLengthXml"
+            files="[\\/]src[\\/]xdocs[\\/](checks|(file)?filters)[\\/](?!.*index\.xml)(?!.*\.xml\.template).+\.xml"/>
```

Comment:
it bad that we put templates and same folder as xdco files.
Lets discuss this in separate issue, please create issue on this.

Reply:
Opened at https://github.com/checkstyle/checkstyle/issues/13512